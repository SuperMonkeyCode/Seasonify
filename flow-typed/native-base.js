/* @flow */
declare module 'native-base' {
  declare var exports: any;
}
