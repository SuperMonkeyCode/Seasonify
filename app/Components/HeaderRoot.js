import React from 'react';
import { Header, Title, Button, Left, Right, Body, Icon } from 'native-base';

export default function HeaderRoot({
  title,
  navigation,
}: {
  title: string,
  navigation: Object,
}) {
  const openDrawer = () => {
    navigation.navigate('DrawerOpen');
  };
  return (
    <Header>
      <Left>
        <Button transparent onPress={openDrawer}>
          <Icon name="menu" />
        </Button>
      </Left>
      <Body>
        <Title>{title}</Title>
      </Body>
      <Right />
    </Header>
  );
}
