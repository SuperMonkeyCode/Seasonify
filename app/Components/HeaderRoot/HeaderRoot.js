/* @flow */
import { Header, Title, Button, Left, Right, Body, Icon } from 'native-base';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  title: string,
  navigation: Object,
  toggleButton?: ReactComponent<*, *, *>,
  /* connect */
  /* HOC */
  handleDrawer: Function,
};

const enhance = compose(
  pure,
  withHandlers({
    handleDrawer: ({ navigation }: Props) => () => {
      navigation.navigate('DrawerOpen');
    },
  }),
);

function HeaderRoot({ title, handleDrawer, toggleButton }: Props) {
  return (
    <Header>
      <Left>
        <Button transparent onPress={handleDrawer}>
          <Icon name="menu" />
        </Button>
      </Left>
      <Body>
        <Title>
          {title}
        </Title>
      </Body>
      <Right>
        {toggleButton || null}
      </Right>
    </Header>
  );
}

export default enhance(HeaderRoot);
