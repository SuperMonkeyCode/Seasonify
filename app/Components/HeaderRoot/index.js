/* @flow */
import HeaderRoot from './HeaderRoot';

export default HeaderRoot;
