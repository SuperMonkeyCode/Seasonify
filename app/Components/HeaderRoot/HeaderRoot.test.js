import { Button, Header, Text } from 'native-base';
import React from 'react';

import HeaderRoot from './HeaderRoot';

function setup(specificProps = {}) {
  const props = {
    title: 'Some tvshow',
    navigation: NavigationMock,
    toggleButton: (
      <Button>
        <Text>toggle</Text>
      </Button>
    ),
    ...specificProps,
  };
  const component = shallowDive(<HeaderRoot {...props} />, Header);
  return {
    component,
    props,
  };
}

describe('Rendering', () => {
  it('should render, with toggle button', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });

  it('should render, WITHOUT toggle button', () => {
    const { component } = setup({ toggleButton: undefined });
    expect(component).toMatchSnapshot();
  });
});

/* ========= Events & Functions ========= */

describe('Events & Functions', () => {
  const { component, props } = setup();
  const backButton = component.find(Button).first();

  describe('when calling onPress on the drawer button', () => {
    backButton.props().onPress();

    it('navigates', () => {
      expect(props.navigation.navigate).toBeCalledWith('DrawerOpen');
    });
  });
});
