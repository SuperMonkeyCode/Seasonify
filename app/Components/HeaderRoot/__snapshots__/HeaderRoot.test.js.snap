// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should render, WITHOUT toggle button 1`] = `
<Styled(Header)>
  <Styled(Left)>
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="menu"
      />
    </Styled(Button)>
  </Styled(Left)>
  <Styled(Body)>
    <Styled(Title)>
      Some tvshow
    </Styled(Title)>
  </Styled(Body)>
  <Styled(Right) />
</Styled(Header)>
`;

exports[`Rendering should render, with toggle button 1`] = `
<Styled(Header)>
  <Styled(Left)>
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="menu"
      />
    </Styled(Button)>
  </Styled(Left)>
  <Styled(Body)>
    <Styled(Title)>
      Some tvshow
    </Styled(Title)>
  </Styled(Body)>
  <Styled(Right)>
    <Styled(Button)>
      <Styled(Text)>
        toggle
      </Styled(Text)>
    </Styled(Button)>
  </Styled(Right)>
</Styled(Header)>
`;
